### DISCUSSION TITLE
Will Renovate ignore an .npmrc file if npmToken is specified in renovate.json?

### DISCUSSION THREAD

**Discussion by grug** [renovatebot/renovate#17652] [Request Help] [answered]
As the title suggests. I have a renovate config file that has the following statement:

```
...
  "encrypted": {
    "npmToken": "<redacted>"
  },
...
```

Will Renovate ignore an `.npmrc` file if it is present in the repository if the above statement is in the renovate config? I've tested things out and it seems that it is indeed the case. I couldn't see this documented anywhere, so just wanted to confirm.

**Comment by rarkins** [comment, marked as answer]
Yes, the npmToken will be converted to npmrc in config, and npmrc in config overrides .npmrc in the repo unless npmrcMerge=true is set

**Reply by grug** [reply]
I've noticed that if the `npmToken` is present in renovate.json, that all other config settings in `.npmrc` are ignored. The one that specifically bit me recently was that `strict-peer-dependencies=false` gets ignored.

**Reply by rarkins** [reply]
https://docs.renovatebot.com/configuration-options/#npmrcmerge

**Reply by grug** [reply]
Looking at the behaviour described in the docs, I think there may be a bug. The docs state:

> If false (default), it means that defining config.npmrc will result in any .npmrc file in the repo being overridden and its values ignored.

However, if you have a renovate.json file that looks like:

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "encrypted": {
    "npmToken": "<your_token_here>"
  }
}
```

and an .npmrc file that looks like:

```
strict-peer-dependencies=false
```

Then the `strict-peer-dependencies=false` value is ignored.

Have I misunderstood something here?

**Reply by rarkins** [reply]
Yes you have, please re-read what I wrote in https://github.com/renovatebot/renovate/discussions/17652#discussioncomment-3568017

**Reply by grug** [reply]
Understood. Is there any chance the documentation can be modified to reflect that it isn't just setting `npmrc` in the renovate config that will cause .npmrc to be ignored? I'm happy to submit a PR if you'd be willing to accept a change like that 😄

**Reply by rarkins** [reply]
Sure, relevant files are https://github.com/renovatebot/renovate/blob/main/docs/usage/configuration-options.md and https://github.com/renovatebot/renovate/blob/main/docs/usage/getting-started/private-packages.md

**Reply by grug** [reply]
Awesome. I really appreciate that. I'll try get a PR up this week.
